### PR TITLE
Rename poi density and use plotly

### LINKED ISSUE #496: POI Indicator - Meta-Issue
- Description is currently specialized on SketchMapTool. Generalize description to encompass POI Layer.
- Rename to “Density”
- should be usable for more topics (i.e. landmarks)
- Change used layer of this Indicator in SketchMapFitness Report to landmarks

### LINKED ISSUE #547: Use plotly in POI density indicator
Use plotly instead of matplotlib in the POI density indicator. Include a svg output for the transition period.

Follow up issue of #489

### PR DESCRIPTION
**Pull request by Gigaszi**
### Description
Rename PoiDensity to Density. Add density indicator to landmarks topic. Use landmarks topic for density indicator in SketchMapFitness report instead of poi topic. Now use plotly to create SVG.

### Corresponding issue
- Closes #496
- Closes #547 

### New or changed dependencies
None

### Checklist
- [x] I have updated my branch to `main` (e.g. through `git rebase main`)
- [x] My code follows the [style guide](https://github.com/GIScience/ohsome-quality-analyst/blob/main/CONTRIBUTING.md#style-guide) and was checked with [pre-commit](https://github.com/GIScience/ohsome-quality-analyst/blob/main/CONTRIBUTING.md#pre-commit) before committing
- [x] I have commented my code
- [ ] I have added sufficient unit and integration [tests](https://github.com/GIScience/ohsome-quality-analyst/blob/main/docs/development_setup.md#tests)
- [x] I have updated the [CHANGELOG.md](https://github.com/GIScience/ohsome-quality-analyst/blob/main/CHANGELOG.md)

### CODE REVIEW

File: workers/tests/integrationtests/fixtures/vcr_cassettes/test_indicator_density.yaml
Comment (joker234) [inline workers/tests/integrationtests/fixtures/vcr_cassettes/test_indicator_density.yaml:1]: I assume the `test_indicator_poi_density.yaml` can be removed now.

File: workers/ohsome_quality_analyst/indicators/density/indicator.py [hunk 1]
```diff
@@ -66,59 +65,96 @@ def create_figure(self) -> None:
...
+                y=[self.count],
+                mode="markers",
+                marker=dict(symbol="circle", size=10, color="black"),
+                name="Location",
+            )
+        )
+
+        # Update layout
+        fig.update_layout(
+            title="Density (POIs per Area)",
```
Comment (joker234) [inline workers/ohsome_quality_analyst/indicators/density/indicator.py:148]:
```suggestion
            title="Density (Features per Area)",
```

File: workers/ohsome_quality_analyst/indicators/density/indicator.py [hunk 2]
```diff
@@ -66,59 +65,96 @@ def create_figure(self) -> None:
...
+            xaxis_title="Area ($km^2$)",
+            yaxis_title="POIs",
```
Comment (joker234) [inline workers/ohsome_quality_analyst/indicators/density/indicator.py:150]:
```suggestion
            yaxis_title="Features",
```

File: workers/ohsome_quality_analyst/indicators/density/metadata.yaml [hunk 3]
```diff
@@ -1,6 +1,6 @@
 ---
-poi-density:
-  name: POI Density
+density:
+  name: Density
   description: >-
       The density of Point of Interests (POI).
```
Comment (joker234) [inline workers/ohsome_quality_analyst/indicators/density/metadata.yaml:5]:
```suggestion
      The density of Features.
```

File: workers/ohsome_quality_analyst/indicators/density/indicator.py [hunk 4]
```diff
@@ -66,59 +65,96 @@ def create_figure(self) -> None:
...
+        raw["layout"].pop("template")  # remove boilerplate
+        self.result.figure = raw
 
-        img_data = StringIO()
-        plt.savefig(img_data, format="svg")
-        self.result.svg = img_data.getvalue()  # this is svg data
-        plt.close("all")
+        # Legacy support for SVGs
+        img_bytes = fig.to_image(format="svg")
+        self.result.svg = img_bytes.decode("utf-8")
```
Comment (joker234) [inline workers/ohsome_quality_analyst/indicators/density/indicator.py:160]: If easily possible, the svg shouldn't have a black background:
test

File: workers/tests/integrationtests/fixtures/vcr_cassettes/test_indicator_density.yaml
Comment (Gigaszi) [reply workers/tests/integrationtests/fixtures/vcr_cassettes/test_indicator_density.yaml:1]: done

File: workers/ohsome_quality_analyst/indicators/density/indicator.py (hunk 1 above)
Comment (Gigaszi) [reply workers/ohsome_quality_analyst/indicators/density/indicator.py]: done

File: workers/ohsome_quality_analyst/indicators/density/indicator.py (hunk 2 above)
Comment (Gigaszi) [reply workers/ohsome_quality_analyst/indicators/density/indicator.py]: done

File: workers/ohsome_quality_analyst/indicators/density/metadata.yaml (hunk 3 above)
Comment (Gigaszi) [reply workers/ohsome_quality_analyst/indicators/density/metadata.yaml]: done

File: workers/ohsome_quality_analyst/indicators/density/indicator.py (hunk 4 above)
Comment (Gigaszi) [reply workers/ohsome_quality_analyst/indicators/density/indicator.py:159]: done

**Review by joker234 — APPROVED** [review]

**Pull request merged**